src/widgets/modals/StocktakeBatchModal.js

```diff
@@ -185,8 +186,24 @@ export const StocktakeBatchModal = ({ stocktakeItem }) => {
   );
 };
 
-StocktakeBatchModal.propTypes = {
+StocktakeBatchModalComponent.propTypes = {
   stocktakeItem: PropTypes.object.isRequired,
+  page: PropTypes.string.isRequired,
+};
+
+const mapStateToProps = state => {
+  const usingPaymentsModule = selectUsingPayments(state);
+  const usingReasons =
+    UIDatabase.objects('NegativeAdjustmentReason').length > 0 &&
+    UIDatabase.objects('PositiveAdjustmentReason').length > 0;
+
+  if (usingReasons) {
+    if (usingPaymentsModule) return { page: 'stocktakeBatchEditModalWithReasonsAndPrices' };
```

Comment:
Just a suggestion as know this is a WIP, but could maybe clean up the long strings with page getters? E.g.

```
page = getPage('stocktakeBatch', 'editModal', { usingReasons, usingPrices });
```

Definitely OTT and would be hard to design elegantly, but a sort of "higher-order" factory/constructor is also a cool option :smiley: , e.g.

```
page = getPage( 'stocktakeBatch', 'editModal').with({ usingReasons, usingPrices });
```

Reply:
Nice - definitely need to come up with a better idea as this clearly isn't working well